tools/ci_build/github/azure-pipelines/binary-size-checks-pipeline.yml

```diff
@@ -0,0 +1,117 @@
+# steps in this build:
+# 1a. Android minimal baseline build (arm64-v8a, including no kernels, with exceptions disabled)
+# 1b. Android minimal baseline build with debug info
+
+parameters:
+- name: DoBuildWithDebugInfo
+  displayName: Create additional build with debug information?
+  type: boolean
+  default: false
+
+jobs:
+- job: BinarySizeChecks
+  timeoutInMinutes: 30
+  workspace:
+    clean: all
+  pool: Linux-CPU-2019
+
+  steps:
+  - checkout: self
+    clean: true
+    submodules: recursive
+
+  - template: templates/get-docker-image-steps.yml
+    parameters:
+      Dockerfile: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cpu
+      Context: tools/ci_build/github/linux/docker
+      DockerBuildArgs: "--build-arg BUILD_UID=$( id -u )"
+      Repository: onnxruntimecpubuild
+
+  - task: CmdLine@2
+    displayName: 1a. Build onnxruntime minimal baseline for Android arm64-v8a
+    inputs:
+      script: |
+        NDK_HOME=$(realpath $ANDROID_NDK_HOME)
+        BINARY_SIZE_THRESHOLD_IN_BYTES=1306224
```

Comment:
Can we set this param using a pipeline variable, so it's easier to update?

Reply:
actually, if we want to specify a limit at all, I think doing so in the code (e.g., here) is probably better since the binary size is directly related to the code. the pipeline may be run on arbitrary commits which have different expected binary size limits.

on the other hand, maybe we don't need to set a hard limit now that we have more consistent binary size reporting.

Reply:
Keep in code is fine, the hard limit is just a warning, but we still want to keep it